src/application_settings/type_notation_helper.py

```diff
@@ -1,17 +1,23 @@
 # pylint: disable=consider-alternative-union-syntax, useless-suppression
 """Defines type aliases that handle notational differences between python versions."""
 import sys
+from collections.abc import Callable
```

Comment:
Why not `typing.Callable`? I think that one is actually included now (through the collections.abc import)

Reply:
It's the other way around: typing.Callable is deprecated (https://docs.python.org/3/library/typing.html#typing.Callable) and that just imports collections.abc.Callable